src/pages/home/report/ReportActionsView.tsx

```diff
@@ -221,6 +224,61 @@ function ReportActionsView({
         }
     }, [hasCachedActions]);
 
+    // When we are offline before opening a money request report,
+    // the total of the report and sometimes the money request aren't displayed because these actions aren't returned until `OpenReport` API is complete.
+    // We generate a fake created action here if it doesn't exist to display the total whenever possible because the total just depends on report data
+    // and we also generate a money request action if If the number of money requests in reportActions is less than the total number of money requests
+    // to display at least one money request action to match the total data.
+    // For more details: https://github.com/Expensify/App/issues/26424#issuecomment-1869154198
+    const reportActionsToDisplay = useMemo(() => {
+        if (!ReportUtils.isMoneyRequestReport(report) || !reportActions.length) {
+            return reportActions;
+        }
+
+        const actions = [...reportActions];
+        const lastAction = reportActions[reportActions.length - 1];
+
+        if (!ReportActionsUtils.isCreatedAction(lastAction)) {
+            const optimisticCreatedAction = ReportUtils.buildOptimisticCreatedReportAction(String(report?.ownerAccountID), DateUtils.subtractMillisecondsFromDateTime(lastAction.created, 1));
+            optimisticCreatedAction.pendingAction = null;
+            actions.push(optimisticCreatedAction);
+        }
+
+        const reportPreviewAction = ReportActionsUtils.getReportPreviewAction(report.chatReportID ?? '', report.reportID);
+        const moneyRequestActions = reportActions.filter(
+            (action) => action.actionName === CONST.REPORT.ACTIONS.TYPE.IOU && action.originalMessage && action.originalMessage.type === CONST.IOU.REPORT_ACTION_TYPE.CREATE,
+        );
+
+        if (report.total && moneyRequestActions.length < (reportPreviewAction?.childMoneyRequestCount ?? 0)) {
+            const optimisticIOUAction = ReportUtils.buildOptimisticIOUReportAction(
+                CONST.IOU.REPORT_ACTION_TYPE.CREATE,
+                0,
+                CONST.CURRENCY.USD,
+                '',
+                [],
+                NumberUtils.rand64(),
+                undefined,
+                report.reportID,
+                false,
+                false,
+                {},
+                false,
+                DateUtils.subtractMillisecondsFromDateTime(actions[actions.length - 1].created, 1),
+            ) as OnyxTypes.ReportAction;
+            moneyRequestActions.push(optimisticIOUAction);
+            actions.splice(actions.length - 1, 0, optimisticIOUAction);
+        }
+
+        // Update pending action of created action if we have some requests that are pending
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        const createdAction = actions.pop()!;
+        if (moneyRequestActions.filter((action) => Boolean(action.pendingAction)).length > 0) {
```

Comment:
@dukenv0307 
I am getting an issue where even in online state, if i can create a request the total has reduced opacity.

This should be case only for offline. Since even the pending requests shows in lesser opacity, the total should behave the same way.


https://github.com/Expensify/App/assets/46707890/e4e9ea25-7ea7-4213-a459-39b0b0beae73

Reply:
Can you try with slow network, it occurs till the money request API succeeds.